File: algorithms/0326.power-of-three/solution.rs
impl Solution {
    pub fn is_power_of_three(n: i32) -> bool {
        let mut n = n;

        if n < 1 {
            return false;
        }

        while n > 0 {
            if n == 1 {
                return true;
            }

            if n % 3 != 0 {
                return false;
            }

            n /= 3;
        }

        false
    }
}

File: algorithms/0326.power-of-three/solution.go
func isPowerOfThree(n int) bool {
	if n < 1 {
		return false
	}

	for n > 0 {
		if n == 1 {
			return true
		}

		if n%3 != 0 {
			return false
		}

		n /= 3
	}

	return false
}
